src/Services/Air/Air.js

```diff
@@ -129,5 +117,55 @@ module.exports = (settings) => {
           err => Promise.reject(new AirRuntimeError.UnableToRetrieveTickets(options, err))
         );
     },
+
+    searchBookingsByPassengerName(options) {
+      const terminal = createTerminalService(settings);
+      return terminal.executeCommand(`*-${options.searchPhrase}`)
+        .then((firstScreen) => {
+          const list = parsers.searchPaxList(firstScreen);
+          if (list) {
+            return Promise
+              .all(list.map((line) => {
+                const localTerminal = createTerminalService(settings);
+                return localTerminal
+                  .executeCommand(`*-${options.searchPhrase}`)
+                  .then((firstScreenAgain) => {
+                    if (firstScreenAgain !== firstScreen) {
+                      return Promise.reject(
+                        new AirRuntimeError.RequestInconsistency({
+                          firstScreen,
+                          firstScreenAgain,
+                        })
+                      );
+                    }
+
+                    return localTerminal.executeCommand(`*${line.id}`);
+                  })
+                  .then(parsers.bookingPnr)
+                  .then(pnr => localTerminal.closeSession()
+                    .then(() => ({ ...line, pnr }))
+                  );
+              }))
+              .then(data => ({ type: 'list', data }));
+          }
+
+          const pnr = parsers.bookingPnr(firstScreen);
+          if (pnr) {
```

Comment:
Once again, I'm voting to return always array.
It will simplify using this call for the 3rd party like this:
```javascript
AirService.searchBookingsByPassengerName({searchPhrase: 'IVANOV'})
  .then(results => results.forEach((item) => { /* do something */ });
```
instead of checking type of response